stubs/Pillow/PIL/ImageFilter.pyi

```diff
@@ -1,115 +1,136 @@
-from typing import Any
+from _typeshed import Self
+from typing import Any, Callable, Sequence, Tuple, Type
+
+from .Image import Image
+
+_FilterArgs = Tuple[Sequence[int], int, int, Sequence[int]]
+
+# filter image parameters below are the C images, i.e. Image().im.
 
 class Filter: ...
 class MultibandFilter(Filter): ...
 
 class BuiltinFilter(MultibandFilter):
-    def filter(self, image): ...
+    def filter(self, image) -> Image: ...
```

Comment:
```suggestion
    def filter(self, image: Image) -> Image: ...
```

Reply:
See my comment here: https://github.com/python/typeshed/pull/5706/files#diff-2c10c136c4c747b4b28ba4bdbe8522a6be30503585842215ad334e0cf9f8138eR8

As far as I can tell, the images here are not `Image`, but the internal C image (aka `Image.im`).

I can certainly fake a class in `_imaging`, then use it in the places where I believe that to be used, though.

Reply:
Oh, good point. Let's not worry about this then, the PR is big enough as is.

Reply:
Yeah, I can send that as another PR later, though. That's further than I went when I originally stubbed out PIL.